From 17b4bb67b0776d76c46b4de4389d1d37faf9231a Mon Sep 17 00:00:00 2001
From: DENG Qingfang <dqfext@gmail.com>
Date: Thu, 16 Mar 2023 19:55:38 +0530
Subject: [PATCH 123/281] OpenWrt:
 711-net-dsa-mv88e6xxx-disable-ATU-violation.patch

net/dsa/mv88e6xxx: disable ATU violation

Signed-off-by: DENG Qingfang <dqfext@gmail.com>

Change-Id: I7e7b3030ea2ddaf2cd7faeed8f443d67e42a04b0
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 drivers/net/dsa/mv88e6xxx/chip.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/dsa/mv88e6xxx/chip.c b/drivers/net/dsa/mv88e6xxx/chip.c
index 0d86ca72459d..09e36a1664a1 100644
--- a/drivers/net/dsa/mv88e6xxx/chip.c
+++ b/drivers/net/dsa/mv88e6xxx/chip.c
@@ -3460,6 +3460,9 @@ static int mv88e6xxx_setup_port(struct mv88e6xxx_chip *chip, int port)
 	else
 		reg = 1 << port;
 
+	/* Disable ATU member violation interrupt */
+	reg |= MV88E6XXX_PORT_ASSOC_VECTOR_IGNORE_WRONG;
+
 	err = mv88e6xxx_port_write(chip, port, MV88E6XXX_PORT_ASSOC_VECTOR,
 				   reg);
 	if (err)
-- 
2.17.1

